src/infrastructure/test/client.py

```diff
@@ -0,0 +1,41 @@
+import urllib.parse
+
+from httpx import AsyncClient, Response
+
+import registry
+from infrastructure.app import create_app
+
+
+class TestClient:
+    def __init__(self):
+
+        app = create_app(registry.routers, registry.middlewares)
+        self.client = AsyncClient(app=app, base_url="http://test.com")
+
+    @staticmethod
+    def _prepare_url(url, query):
+        return f"{url}{urllib.parse.urlencode(query)}"
+
+    async def post(self, url, query=None, data=None, headers=None) -> Response:
+        if query:
+            url = self._prepare_url(url, query)
+        response = await self.client.post(url, data=data, headers=headers)
+        return response
+
+    async def put(self, url, query=None, data=None, headers=None):
+        if query:
+            url = self._prepare_url(url, query)
+        response = await self.client.put(url, data=data, headers=headers)
+        return response
+
+    async def get(self, url, query=None, headers=None):
+        if query:
+            url = self._prepare_url(url, query)
+        response = await self.client.get(url, headers=headers)
+        return response
+
+    async def delete(self, url, query=None, headers=None):
+        if query:
+            url = self._prepare_url(url, query)
+        response = await self.client.delete(url, headers=headers)
+        return response
```

Comment:
I would recommend using conftest instead of that separate class and also using FastAPI already has this kind of client out of the box

```python
from fastapi.testclient import TestClient

@pytest.fixture()
def client(app):
    yield TestClient(app)
```

Check this one - https://fastapi.tiangolo.com/advanced/async-tests/

Reply:
Using in this way is correct way, but I am sure we will add some additional methods to skip auth methods, or such steps.

Reply:
We can replace deps if we need to achieve that

https://fastapi.tiangolo.com/advanced/testing-dependencies/#use-the-appdependency_overrides-attribute